scrapy/utils/python.py

```diff
@@ -254,25 +254,22 @@ def get_spec(func):
 def equal_attributes(obj1, obj2, attributes):
     """Compare two objects attributes"""
     # not attributes given return False by default
-    if not attributes:
-        return False
```

Comment:
This is a common style to write functions: first we check special cases, and then we write the main function body. It allows to remove unnecessary nesting and makes code easier to read to people familiar with this convention. I.e. in existing implementation it reads like the main thing function is doing is checking each attribute (`for attr in attributes:`); in your version this loop is in an indented block after some statements. It may be a personal preference, but I like the existing way to write it more.